src/main/java/org/broadinstitute/hellbender/tools/sv/cluster/CanonicalSVCollapser.java

```diff
@@ -374,22 +414,203 @@ private static int compareCopyNumberForCollapsing(final Integer first, final Int
+        }
+    }
+
+    /**
+     * Generates genotype alleles for deletion genotypes from the given copy number.
+     * @param refAllele  reference allele for the site
+     * @param expectedCopyNumber  expected copy number for the genotype
+     * @param copyNumber  copy number for the genotype
+     * @return  genotype alleles
+     */
+    public static List<Allele> getDeletionAllelesFromCopyNumber(final Allele refAllele, final int expectedCopyNumber,
+                                                                final int copyNumber) {
+        Utils.nonNull(refAllele);
+        Utils.validateArg(copyNumber <= expectedCopyNumber, "Invalid deletion copy number " +
+                copyNumber + " when ploidy is " + expectedCopyNumber);
+        Utils.validateArg(expectedCopyNumber >= 0, "Invalid expected copy number: " + expectedCopyNumber);
+        if (expectedCopyNumber == 0) {
+            return Collections.emptyList();
+        } else if (expectedCopyNumber == copyNumber) {
+            // Most common in practice - use faster method
+            return Collections.nCopies(expectedCopyNumber, refAllele);
+        } else {
+            final List<Allele> alleles = new ArrayList<>(expectedCopyNumber);
+            for (int i = 0; i < copyNumber; i++) {
+                alleles.add(refAllele);
+            }
+            for (int i = copyNumber; i < expectedCopyNumber; i++) {
+                alleles.add(Allele.SV_SIMPLE_DEL);
+            }
+            return alleles;
+        }
+    }
+
+    /**
+     * Generates genotype alleles for duplication genotypes from the given copy number. Genotypes that cannot be
+     * determined unambiguously (e.g. diploid sites) result in {@link Allele#NO_CALL} alleles.
+     * @param refAllele  reference allele for the site
+     * @param expectedCopyNumber  expected copy number for the genotype
+     * @param copyNumber  copy number for the genotype
+     * @return  genotype alleles
+     */
+    public static List<Allele> getDuplicationAllelesFromCopyNumber(final Allele refAllele, final int expectedCopyNumber,
+                                                                   final int copyNumber) {
+        Utils.nonNull(refAllele);
+        Utils.validateArg(copyNumber >= expectedCopyNumber, "Invalid duplication copy number " +
+                copyNumber + " when ploidy is " + expectedCopyNumber);
+        Utils.validateArg(expectedCopyNumber >= 0, "Invalid expected copy number: " + expectedCopyNumber);
+        if (expectedCopyNumber == copyNumber) {
+            // Most common in practice - use faster method
+            return Collections.nCopies(expectedCopyNumber, refAllele);
+        } else if (expectedCopyNumber == 0) {
+            return Collections.emptyList();
+        } else if (expectedCopyNumber == 1) {
+            // At this point know it's a DUP, and since it's haploid the phasing is unambiguous
```

Comment:
This seems like the right thing to do given how we are using the output VCF downstream, but it's unfortunate that the allele is named `SV_SIMPLE_DUP` which implies a single duplication of the segment. I'm not sure how best to make it less confusing, though.